maserol/preprocess.py

```diff
@@ -29,14 +31,35 @@ def prepare_data(data: xr.DataArray, remove_rcp=None):
         data = data.rename({"Subject": "Sample"})
     data = data.transpose("Sample", "Receptor", "Antigen")
 
-    # Receptors: only keep those with "IgGx" or "FcgRx"
-    data_receptors = data.Receptor.values
-    wanted_receptors = [x for x in data_receptors if re.match("^igg", x, flags=re.IGNORECASE)] + \
-                       [x for x in data_receptors if re.match("fc[gr]*", x, flags=re.IGNORECASE) and x != "FcRalpha"]
-    if remove_rcp != None:
-        for r in remove_rcp:
-            wanted_receptors.remove(r)
-    data = data.sel(Receptor=wanted_receptors)
+    wanted_receptors = data.Receptor.values
+    if data_id is None:
+        # Receptors: only keep those with "IgGx" or "FcgRx"
+        wanted_receptors = [x for x in wanted_receptors if re.match("^igg", x, flags=re.IGNORECASE)] + \
+                           [x for x in wanted_receptors if re.match("fc[gr]*", x, flags=re.IGNORECASE) and x != "FcRalpha"]
+        if remove_rcp != None:
+            for r in remove_rcp:
+                wanted_receptors.remove(r)
+        data = data.sel(Receptor=wanted_receptors)
+    else:
```

Comment:
You probably have a good reason for setting this up, but it's also helpful to keep things simple

Reply:
The alternative is specifying this information each time a dataset is loaded, so I think the added complexity is worth it (this logic is pretty generic, so I'm guessing the yaml file will be added to in the future).